fuuk/people/admin/options.py

```diff
@@ -147,39 +149,118 @@ class AgencyAdmin(TranslationAdmin):
     search_fields = get_translation_fields('name') + get_translation_fields('shortcut')
 
 
+class GrantCollaboratorInlineAdmin(TabularInline):
+    """
+    `GrantCollaborator` objects can be added only through GrantAdmin. Each user who can change the grant can modifie
+    attached GrantCollaborators too. `GrantCollaboratorInlineFormSet` checks if `Grant.investigator_human` is not among
+    `GrantCollaborator.human` objects.
+    """
+    formset = GrantCollaboratorInlineFormSet
+    model = GrantCollaborator
+    extra = 3
+
+    def has_add_permission(self, request):
+        return True
+
+    def has_change_permission(self, request, obj=None):
+        return True
+
+    def has_delete_permission(self, request, obj=None):
+        return True
+
+
 class GrantAdmin(TranslationAdmin):
-    list_display = ('author', 'number', 'title', 'start', 'end')
+    list_display = ('investigator_name', 'number', 'title', 'start', 'end')
     list_filter = ('agency', 'start')
     ordering = ('-end', )
     search_fields = ['number'] + get_translation_fields('title')
-    filter_horizontal = ('co_authors',)
+    inlines = (GrantCollaboratorInlineAdmin, )
+
+    def get_readonly_fields(self, request, obj=None):
+        """
+        `editor` can be changed only by the user with the change permission for grants.
+        """
+        if super(GrantAdmin, self).has_change_permission(request, obj):
```

Comment:
The change permission should be checked directly. Using `super` in this case may introduce unexpected problems.